bundles/org.eclipse.passage.lic.oshi/src/org/eclipse/passage/lic/oshi/FragileData.java

```diff
@@ -28,21 +28,18 @@
 final class FragileData<T> {
 
 	private final Supplier<T> aspect;
-	private final Consumer<T> read;
+	private final Consumer<Supplier<T>> read;
 
-	FragileData(Supplier<T> aspect, Consumer<T> read) {
+	FragileData(Supplier<T> aspect, Consumer<Supplier<T>> read) {
 		this.aspect = aspect;
 		this.read = read;
 	}
 
 	void supply() {
-		T descriptor;
 		try {
-			descriptor = aspect.get();
+			read.accept(aspect);
```

Comment:
I want to suggest to replace this entire class by a try-catch-block in the two places that use it.
With this change it lead to IMHO more readable code, especially because `Swath` only creates the lambdas to pass it to this class.